### PR TITLE
Fix macos build and support multiple files

### PR DESCRIPTION
**Pull request by justin808**
Conversion of template yaml to hash and back resulted in the loss of multiple documents in one yaml file.

MacOS builds require specification of the platform.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -208,6 +232,61 @@ apps:
     dockerfile: ../some_other/Dockerfile
 ```
 
+## Workflow
```
Comment (rafaelgomesxyz) [inline README.md:235]: By the way, this flow is already mostly documented in https://github.com/shakacode/heroku-to-control-plane/blob/main/docs/migrating.md#clone-the-staging-environment. Do we want to duplicate it here?

File: README.md [hunk 2]
```diff
@@ -89,21 +89,23 @@ For the typical Rails app, this means:
...
-   For information on the latter, see [CONTRIBUTING.md](CONTRIBUTING.md).
+1. Run `cpln image docker-login --org <your-org>` to ensure that you have access to the Control Plane Docker registry.
 
-```sh
-gem install cpl
-```
+1. Install Heroku to Control Plane `cpl` CLI, either as a [Ruby gem](https://rubygems.org/gems/cpl) or a local clone.
+   For information on the latter, see [CONTRIBUTING.md](CONTRIBUTING.md). You may also install `cpl` in your project's Gemfile.
+
+1. This project has a `Dockerfile` for Control Plane in this directory. You can use it as an example for your project. Ensure that you have Docker running.
```
[the remark this answers is not in the document]
Comment (ahangarha) [reply README.md:108]: This is coming from RWRT project

[the remark this answers is not in the document]
Comment (rafaelgomesxyz) [reply README.md:108]: Ok, we should specify then and add a link to the `Dockerfile` there.

File: README.md [hunk 3]
```diff
@@ -89,21 +89,23 @@ For the typical Rails app, this means:
 
 ## Installation
 
+1. Ensure your [Control Plane](https://controlplane.com) account is set up. Set up an `organization` <your-org> for testing in that account and modify value for `aliases.common.cpln_org` in `.controlplane/controlplane.yml`. If you need an organization, please [contact Shakcode](mailto:controlplane@shkacode.com).
```
[the remark this answers is not in the document]
Comment (justin808) [reply README.md:92]: nice catch!

File: README.md [hunk 4]
```diff
@@ -89,21 +89,23 @@ For the typical Rails app, this means:
...
 1. Install [Node.js](https://nodejs.org/en) (required for Control Plane CLI).
-2. Install [Ruby](https://www.ruby-lang.org/en/) (required for these helpers).
-3. Install Control Plane CLI (adds `cpln` command) and configure credentials.
 
+1. Install [Ruby](https://www.ruby-lang.org/en/) (required for these helpers).
+
+1. Install Control Plane CLI (and configure access) [docs here](https://docs.controlplane.com/quickstart/quick-start-3-cli#getting-started-with-the-cli), `npm install -g @controlplane/cli`. You can update the `cpln` command line with `npm update -g @controlplane/cli`. Then run `cpln login` to ensure access.
```
[the remark this answers is not in the document]
Comment (justin808) [reply README.md:97]: yes, feel free to clean up

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (justin808) [reply README.md:108]: yes -- copy paste error from rwrt project, and yes, we can link there.

File: README.md [hunk 5]
```diff
@@ -112,10 +114,32 @@ The `cpln` CLI is the Control Plane CLI.
 
 Click [here](/docs/migrating.md) to see the steps to migrate.
 
-## Config
+## Configuration Files
 
-Here's a complete example of all supported config keys explained:
+The `cpl` gem is based on several configuration files within a `/.controlplane` top-level directory in your Rails project.
```
[the remark this answers is not in the document]
Comment (justin808) [reply README.md:119]: yes -- remove word `Rails`

File: README.md [hunk 6]
```diff
@@ -208,6 +232,61 @@ apps:
...
+```sh
+# Provision all infrastructure on Control Plane.
+# app tutorial-app will be created per definition in .controlplane/controlplane.yml
+cpl apply-template gvc postgres redis rails -a tutorial-app
+
+# Build and push docker image to Control Plane repository
+# Note, may take many minutes. Be patient. Check for error messages, such as forgetting to run `cpln image docker-login --org <your-org>`
+cpl build-image -a tutorial-app
+
+# Promote image to app after running `cpl build-image command`
```
[the remark this answers is not in the document]
Comment (justin808) [reply README.md:252]: yes, good catch

File: README.md [hunk 7]
```diff
@@ -208,6 +232,61 @@ apps:
...
+After committing code, you will update your deployment of `tutorial-app` with the following commands:
+
+```sh
+# Build and push new image with sequential image tagging, e.g. 'tutorial-app:1', then 'tutorial-app:2', etc.
+cpl build-image -a tutorial-app
+
+# Run database migrations (or other release tasks) with latest image,
+# while app is still running on previous image.
+# This is analogous to the release phase.
+cpl runner rails db:migrate -a tutorial-app --image latest
```
[the remark this answers is not in the document]
Comment (justin808) [reply README.md:274]: yes -- that needs to be fixed in the RWRT project as well @ahangarha

[the remark this answers is not in the document]
Comment (justin808) [reply README.md:274]: btw, should the command always be run detached? @dzirtusss ?

File: README.md [hunk 8]
```diff
@@ -208,6 +232,61 @@ apps:
...
+
+If you needed to push a new image with a specific commit SHA, you can run the following command:
+
+```sh
+# Build and push with sequential image tagging and commit SHA, e.g. 'tutorial-app:123_ABCD'
+cpl build-image -a tutorial-app --commit ABCD
+```
+
+### Real World
+Most companies will configure their CI system to handle the above steps. Please [contact Shakcode](mailto:controlplane@shkacode.com) for examples of how to do this, jump on [**React+Rails Slack channel**] (https://reactrails.slack.com/join/shared_invite/enQtNjY3NTczMjczNzYxLTlmYjdiZmY3MTVlMzU2YWE0OWM0MzNiZDI0MzdkZGFiZTFkYTFkOGVjODBmOWEyYWQ3MzA2NGE1YWJjNmVlMGE).
```
[the remark this answers is not in the document]
Comment (justin808) [reply README.md:288]: yes, please fix!

[the remark this answers is not in the document]
Comment (justin808) [reply README.md:288]:
```
Most companies will configure their CI system to handle the above steps. Please [contact Shakcode](mailto:controlplane@shakacode.com) for examples of how to do this, or jump on [**React+Rails Slack channel**] ```

File: README.md (hunk 1 above)
Comment (justin808) [reply README.md:235]: The migration should probably refer to this.

We need to let people know about the simple flow without a full migration.

**Comment by justin808** [comment]
@rafaelgomesxyz ready to merge?

**Comment by rafaelgomesxyz** [comment]
> @rafaelgomesxyz ready to merge?

From my side, yes.

**Comment by justin808** [comment]
thank you @rafaelgomesxyz!

**Pull request merged**